capstone/src/main/java/com/google/sps/servlets/clubs/LabelsServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/* Servlet that stores and returns data relating to clubs. */
+@WebServlet("/labels")
+public class LabelsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Map<String, Integer> labelFrequency = new HashMap<String, Integer>();
+    List<String> labels = new ArrayList<String>();
+
+    Query query = new Query(Constants.CLUB_ENTITY_PROP);
+    PreparedQuery prepared = datastore.prepare(query);
+
+    for (Entity entity : prepared.asIterable()) {
+      for (String label : ServletUtil.getPropertyList(entity, Constants.LABELS_PROP)) {
+        if (labelFrequency.keySet().contains(label)) {
```

Comment:
How are clubs adding labels? Are they free-hand typing into a box where they just make a comma-separated list? It seems so but maybe I missed something.

If that is the case you might want to do some cleanup:
(1) Make all lowercase (so `Sports` is the same as `sports` and `SPORTS`) 
(2) Remove any spacing at the beginning and end (so list `sports, culture, hobby' is parsed like 'sports,culture,hobby')

Would probably add a test to make sure this is happening properly as well.

Reply:
Sounds good - I'm thinking about actually removing all whitespace (so multiword labels would need to be like best-club).